aws/data_source_aws_vpc_endpoint_service.go

```diff
@@ -69,25 +69,40 @@ func dataSourceAwsVpcEndpointService() *schema.Resource {
 				Type:     schema.TypeBool,
 				Computed: true,
 			},
+			"filter": dataSourceFiltersSchema(),
 		},
 	}
 }
 
 func dataSourceAwsVpcEndpointServiceRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ec2conn
 
+	filters, filtersOk := d.GetOk("filter")
+	tags, tagsOk := d.GetOk("tags")
+
 	var serviceName string
+	serviceNameOk := false
 	if v, ok := d.GetOk("service_name"); ok {
 		serviceName = v.(string)
+		serviceNameOk = true
 	} else if v, ok := d.GetOk("service"); ok {
 		serviceName = fmt.Sprintf("com.amazonaws.%s.%s", meta.(*AWSClient).region, v.(string))
-	} else {
-		return fmt.Errorf(
-			"One of ['service', 'service_name'] must be set to query VPC Endpoint Services")
+		serviceNameOk = true
+	}
+
+	if !filtersOk && !serviceNameOk && !tagsOk {
```

Comment:
Can this now be done at `terraform plan` time with [`Schema.AtLeastOneOf`](https://github.com/hashicorp/terraform-plugin-sdk/blob/ac2f8d01136a1a0198e27d49ea0ea6ed986ea4ee/helper/schema/schema.go#L215-L228)?

Reply:
let me check this

Reply:
Or maybe we shouldn't have this (and other ones in other data sources) at all?
It seems many other EC2 data sources don't have such checks and leave it to the user to ensure that enough parameters are supplied to ensure that just 1 resource matches.

Reply:
i took this from data_aws_instance. this is the closest to plan validation as `tags` and `filters` are functions.

Reply:
I thought about it, removing it seems fine. will the aws provided error be descript enough in your opinion?

Reply:
we shouldn't hit an AWS error I don' think.
If there are more than 1 endpoint services that match the parameters (if any) then we will hit the `return fmt.Errorf("multiple VPC Endpoint Services matched; use additional constraints to reduce matches to a single VPC Endpoint Service")`.

Reply:
gotcha, i'll remove it from both PRs.

Reply:
removed & tested.